float32[] features
---
bool[] val